packages/flutter/lib/src/material/ink_sparkle.dart

```diff
@@ -516,12 +501,11 @@ double _getTargetRadius(
 class FragmentShaderManager {
   FragmentShaderManager._();
 
-  static late ui.FragmentProgram _program;
+  static final ui.FragmentProgram _program = ui.FragmentProgram.fromAsset('shaders/ink_sparkle.frag');
```

Comment:
This `FragmentProgram.fromAsset()` call takes ~9ms on my 2015 OnePlus 2. I'm wondering if `fromAsset()` should return a `Future` to force it to happen outside of frame workloads. Wdyt?

Reply:
A few thoughts:

1. How expensive would this be in impeller? If its just the asset loading, we may not need the future, but we can always add new APIs later.

2. I don't really love the strategy of forcing a Future or microtask in dart:ui. That can still block by delaying the next frame if the UI thread would be quite busy. If these compilations are going to be that expensive on low end devices, maybe we should push compilation itself to a worker thread instead. That would also have the benefit of avoiding the poor interaction of the flutter test framework and the non-microtask `Future` constructors.

Reply:
We have time before the next stable to benchmark on Impeller and then decide what to do. I expect the overhead to be lower, though, since the shader will already be lowered/compiled for OpenGL/Vulkan/Metal, and so we'd only be paying for the asset load and maybe a copy.

Reply:
From discussion offline: I'm going to back out the ink sparkle specific changes from this PR. I need to investigate:

1. Does the scheduling optimization we have work with shader compilation, or does it still lead to jank.
2. Is the expensive part of shader creation on skia actually constructing the SkRuntimeEffect, or does work get deferred until we use it.
3. How expensive is this operation on impeller

Reply:
So the answer to: does Skia do expensive work when constructing an SkRuntimeEffect is, yes - that is where the actual program compilation appears to happen:

![image](https://user-images.githubusercontent.com/8975114/181105651-740af6fd-6100-4789-866a-e12c9558c704.png)

And also: https://github.com/google/skia/blob/main/src/core/SkRuntimeEffect.cpp#L330

Reply:
Another thought - to support this on the web we'd essentially have to make fromAsset async (since it will need to make a network request). So why don't we make this async and then if we get to a world where it is always super fast, we can add a sync version?